### PR TITLE
Make very minor changes to extension philosophy section

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: docs/src/main/asciidoc/extension-authors-guide.adoc [hunk 1]
```diff
@@ -23,18 +23,18 @@ This section is a work in progress and gathers the philosophy under which extens
 
 === Why an extension framework
 
-Quarkus’s mission is to transform your full application and the libraries it uses into an optimal output for GraalVM.
+Quarkus' mission is to transform your entire application including the libraries it uses, into an artifact optimized for GraalVM.
```
Comment (emmanuelbernard) [inline docs/src/main/asciidoc/extension-authors-guide.adoc]: I had it as `Quarkus’` initially and @LightGuard corrected it to `Quarkus’s`
Let's not change it

File: docs/src/main/asciidoc/extension-authors-guide.adoc [hunk 2]
```diff
@@ -23,18 +23,18 @@ This section is a work in progress and gathers the philosophy under which extens
...
 To do this you need to analyze and understand the full "closed world" of the application.
 Without the full and complete context, the best that can be achieved is partial and limited generic support.
 By using the Quarkus extension approach, we can bring Java applications in line with memory footprint constrained environments like Kubernetes or cloud platforms.
 
-The Quarkus extension framework makes a significant impact even when GraalVM is not used (e.g. in HotSpot). 
-Let’s list the actions an extension does:
+It's important to note that the Quarkus extension framework results in significantly improved resource utilization even when GraalVM is not used (e.g. in HotSpot).
```
Comment (emmanuelbernard) [inline docs/src/main/asciidoc/extension-authors-guide.adoc]: It's important to note that brings nothing but weight to the sentence. Can you remove it?

**Review by emmanuelbernard — CHANGES_REQUESTED** [review]
Thanks for the improvement. I added a few comments. Can you apply then and it's good to go.

File: docs/src/main/asciidoc/extension-authors-guide.adoc (hunk 2 above)
Comment (geoand) [reply docs/src/main/asciidoc/extension-authors-guide.adoc]: :+1: , done

File: docs/src/main/asciidoc/extension-authors-guide.adoc (hunk 1 above)
Comment (geoand) [reply docs/src/main/asciidoc/extension-authors-guide.adoc]: Done

**Comment by geoand** [comment]
@emmanuelbernard PR updated as per your comments :)

**Review by emmanuelbernard — APPROVED** [review]

**Pull request merged**